src/routes/categories/containers/__tests__/LanguageFailure.spec.js

```diff
@@ -0,0 +1,147 @@
+  ]
+
+  const language = 'en'
+
+  test('should match snapshot', () => {
+    const wrapper = shallow(
+      <LanguageFailure locations={locations}
+                       languages={languages}
+                       location={location}
+                       language={language}
+                       t={key => key}
+                       setLanguageChangeUrls={() => {}} />
+    )
+
+    expect(wrapper).toMatchSnapshot()
+  })
+
+  test('should dispatch once on mount', () => {
+    const mockSetLanguageChangeUrls = jest.fn()
+    const categoriesPage = shallow(
+      <LanguageFailure locations={locations}
+                       languages={languages}
+                       location={location}
+                       language={language}
+                       setLanguageChangeUrls={mockSetLanguageChangeUrls}
+                       t={key => key} />
+    ).instance()
+
+    expect(mockSetLanguageChangeUrls.mock.calls).toHaveLength(1)
+    expect(mockSetLanguageChangeUrls).toBeCalledWith(categoriesPage.mapLanguageToUrl, languages)
+  })
+
+  test('should map language to url', () => {
+    const mapLanguageToUrl = shallow(
+      <LanguageFailure locations={locations}
+                       languages={languages}
+                       location={location}
+                       language={language}
+                       setLanguageChangeUrls={() => {}}
+                       t={key => key} />
+    ).instance().mapLanguageToUrl
+
+    expect(mapLanguageToUrl(language)).toBe(`/${location}/${language}`)
+  })
+
+  describe('connect', () => {
+    const languagesEndpoint = new EndpointBuilder('languages')
+      .withUrl('https://weird-endpoint/api.json')
+      .withMapper(json => json)
+      .withResponseOverride(languages)
+      .build()
+
+    const locationsEndpoint = new EndpointBuilder('locations')
+      .withUrl('https://weird-endpoint/api.json')
+      .withMapper(json => json)
+      .withResponseOverride(locations)
+      .build()
+
+    const pathname = '/augsburg/tu'
+    const id = '1234'
+
+    test('should map state to props', () => {
+      const store = createReduxStore(createHistory, {
+        router: {
+          params: {location: location, language: language},
+          pathname: pathname,
+          query: {id: id}
+        },
+        languageChangeUrls: {}
+      })
+
+      const languageFailure = mount(
+        <Provider store={store}>
+          <EndpointProvider endpoints={[locationsEndpoint, languagesEndpoint]}>
+            <ConnectedLanguageFailure />
+          </EndpointProvider>
+        </Provider>
+      ).find(LanguageFailure)
+
+      expect(languageFailure.props()).toEqual({
+        location: location,
+        setLanguageChangeUrls: expect.any(Function),
+        t: expect.any(Function),
+        locations: locations,
+        languages: languages
+      })
+    })
+
+    test('should map dispatch to props', () => {
+      const store = createReduxStore(createHistory, {
```

Comment:
I think a mockstore should be enough here because you are only checking if the setLanguageChangeUrls action has been called.

Reply:
Since LanguageFailure has two withFetcher prepended, the standard mockStore doesn't work here.